src/actions/mintNFT.ts

```diff
@@ -1,3 +1,152 @@
+import { MintLayout, TOKEN_PROGRAM_ID, Token } from '@solana/spl-token';
+import { Keypair, PublicKey } from '@solana/web3.js';
+import { Connection } from './../Connection';
+import { MintTo, CreateAssociatedTokenAccount, CreateMint, Transaction } from './../programs';
+import {
+  CreateMasterEdition,
+  CreateMetadata,
+  Creator,
+  MasterEdition,
+  Metadata,
+  MetadataDataData,
+} from './../programs/metadata';
+import { Wallet } from 'src/wallet';
+import { ASSOCIATED_TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import { sendTransaction } from './transactions';
+import { lookup } from './../utils/metadata';
+import BN from 'bn.js';
+
+interface MintNFTParams {
+  connection: Connection;
+  wallet: Wallet;
+  uri: string;
+  maxSupply: number;
+}
+
+interface MintNFTResponse {
+  txId: string;
+  mint: PublicKey;
+  metadata: PublicKey;
+  edition: PublicKey;
+}
+
+export const mintNFT = async ({
+  connection,
+  wallet,
+  uri,
+  maxSupply,
+}: MintNFTParams): Promise<MintNFTResponse> => {
+  const mint = Keypair.generate();
+
+  const metadataPDA = await Metadata.getPDA(mint.publicKey);
+  const editionPDA = await MasterEdition.getPDA(mint.publicKey);
+
+  const mintRent = await connection.getMinimumBalanceForRentExemption(MintLayout.span);
+
+  const {
+    name,
+    symbol,
+    seller_fee_basis_points,
+    properties: { creators },
+  } = await lookup(uri);
+
+  const creatorsData = creators.reduce((memo, { address, share }) => {
+    // QUESTION: setting verified to true when creator address matches the wallet creating the NFT. OK?
+    const verified = address === wallet.publicKey.toString();
+
+    const creator = new Creator({
+      address,
+      share,
+      verified,
+    });
+
+    memo = [...memo, creator];
+
+    return memo;
+  }, [] as Creator[]);
+
+  const createMintTx = new CreateMint(
+    { feePayer: wallet.publicKey },
+    {
+      newAccountPubkey: mint.publicKey,
+      lamports: mintRent,
+    },
+  );
+
+  const metadataData = new MetadataDataData({
+    name,
+    symbol,
+    uri,
+    sellerFeeBasisPoints: seller_fee_basis_points,
+    creators: creatorsData,
+  });
+
+  const createMetadataTx = new CreateMetadata(
+    {
+      feePayer: wallet.publicKey,
+    },
+    {
+      metadata: metadataPDA,
+      metadataData,
+      updateAuthority: wallet.publicKey,
+      mint: mint.publicKey,
+      mintAuthority: wallet.publicKey,
+    },
+  );
+
+  const recipient = await Token.getAssociatedTokenAddress(
+    ASSOCIATED_TOKEN_PROGRAM_ID,
+    TOKEN_PROGRAM_ID,
+    mint.publicKey,
+    wallet.publicKey,
+  );
+
+  const createAssociatedTokenAccountTx = new CreateAssociatedTokenAccount(
+    { feePayer: wallet.publicKey },
+    {
+      associatedTokenAddress: recipient,
+      splTokenMintAddress: mint.publicKey,
+    },
+  );
+
+  const mintToTx = new MintTo(
+    { feePayer: wallet.publicKey },
+    {
+      mint: mint.publicKey,
+      dest: recipient,
+      amount: 1,
```

Comment:
I guess amount should be configurable

Reply:
I'm not that aware of the flow, but can we mint a few items? And does it makes sense?

Reply:
No, let's stick to 1 for now